pkg/webhooks/node/node.go

```diff
@@ -0,0 +1,166 @@
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+
+	return valid
+}
+
+// Authorized implements Webhook interface
+func (s *NodeWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	return s.authorized(request)
+}
+
+func (s *NodeWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	if request.AdmissionRequest.UserInfo.Username == "system:unauthenticated" {
+		// This could highlight a significant problem with RBAC since an
+		// unauthenticated user should have no permissions.
+		log.Info("system:unauthenticated made a webhook request. Check RBAC rules", "request", request.AdmissionRequest)
+		ret = admissionctl.Denied("Unauthenticated")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+	if strings.HasPrefix(request.AdmissionRequest.UserInfo.Username, "system:") {
+		ret = admissionctl.Allowed("authenticated system: users are allowed")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+	if strings.HasPrefix(request.AdmissionRequest.UserInfo.Username, "kube:") {
+		ret = admissionctl.Allowed("kube: users are allowed")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+	if slices.Contains(adminUsers, request.AdmissionRequest.UserInfo.Username) {
+		ret = admissionctl.Allowed("Specified admin users are allowed")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+	for _, userGroup := range request.UserInfo.Groups {
+		if slices.Contains(adminGroups, userGroup) {
+			ret = admissionctl.Allowed("Members of admin groups are allowed")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	if isProtectedActionOnNode(request) {
+		localmetrics.IncrementNodeWebhookBlockedRequest(request.UserInfo.Username)
+
+		ret = admissionctl.Denied(`Prevented from accessing Red Hat managed resources. This is an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support.
+
+To modify node labels or taints, use OCM or the ROSA cli to edit the MachinePool.`)
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+
+	log.Info("Denying access", "request", request.AdmissionRequest)
+	ret = admissionctl.Denied("Prevented from accessing Red Hat managed resources. This is in an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// isProtectedActionOnNode checks if the request is to update a node
+func isProtectedActionOnNode(request admissionctl.Request) bool {
+	return request.Kind.Kind == "Node" &&
+		(request.Operation == admissionv1.Update || request.Operation == admissionv1.Create || request.Operation == admissionv1.Delete)
```

Comment:
```suggestion
	return request.Kind.Kind == "Node"
```
Sorry, for a bit of whiplash on this, but I think a check like this should be sufficient and we can gate this using what's currently on line 34 to adjust the "protected actions" there

```go
			Operations: []admissionregv1.OperationType{"*"},
```